src/ethereum/berlin/vm/memory.py

```diff
@@ -80,4 +85,27 @@ def memory_read_bytes(
     data_bytes :
         Data read from memory.
     """
-    return memory[start_position : start_position + size]
+    return memory[start_position : Uint(start_position) + Uint(size)]
+
+
+def buffer_read(buffer: Bytes, start_position: U256, size: U256) -> Bytes:
```

Comment:
Naming the `buffer_read()` function is hard. Geth has the impressively vague:

```
// getData returns a slice from the data based on the start and size and pads
// up to size with zero's. This function is overflow safe.
func getData(data []byte, start uint64, size uint64) []byte {
```

We could use `buffer_read()` for memory. The padding is never used for memory because memory extension guarantees that memory reads will be in bounds unless the length is zero.

Reply:
> Naming the `buffer_read()` function is hard.

Agreed. This seems like a decent PR to fix it in though.